apis/core/lib/csvw-builder/index.ts

```diff
@@ -48,22 +48,36 @@ function mappedLiteralColumn({ csvMapping, columnMapping, column }: CsvwBuilding
   return csvwColumn
 }
 
-async function mappedReferenceColumn({ csvMapping, columnMapping, resources }: CsvwBuildingContext & { columnMapping: ReferenceColumnMapping }): Promise<Initializer<Csvw.Column>> {
+async function mappedReferenceColumn({ csvMapping, columnMapping, source, resources }: CsvwBuildingContext & { columnMapping: ReferenceColumnMapping }): Promise<Initializer<Csvw.Column>> {
   const csvwColumn: Initializer<Csvw.Column> = {
     propertyUrl: csvMapping.createIdentifier(columnMapping.targetProperty).value,
   }
 
+  const referencedTable = await resources.getResource<Table>(columnMapping.referencedTable.id)
+  if (!referencedTable) {
+    throw new Error(`Table ${columnMapping.referencedTable.id} not found`)
+  }
+
+  if (!referencedTable.csvSource) {
+    throw new Error(`Table ${columnMapping.referencedTable.id} not linked to a source`)
+  }
+
+  const referencedSource = await resources.getResource<CsvSource>(referencedTable.csvSource.id)
+  if (!referencedSource) {
+    throw new Error(`Source ${referencedTable.csvSource.id} not found`)
+  }
```

Comment:
If you rebase now you'll be able to remove the throw

```suggestion
  const referencedTable = await resources.getResource<Table>(columnMapping.referencedTable.id)

  if (!referencedTable.csvSource) {
    throw new Error(`Table ${columnMapping.referencedTable.id} not linked to a source`)
  }

  const referencedSource = await resources.getResource<CsvSource>(referencedTable.csvSource.id)
```